components/carousel/index.jsx

```diff
@@ -0,0 +1,287 @@
+import AutoPlayButton from './private/auto-play-button';
+
+// ### Event Helpers
+import KEYS from '../../utilities/key-code';
+import EventUtil from '../../utilities/event';
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/**
+ * A carousel allows multiple pieces of featured content to occupy an allocated amount of space.
+ */
+class Carousel extends React.Component {
+	constructor(props) {
+		super(props);
+
+		const { items, itemsPerPanel } = this.props;
+
+		this.generatedId = shortid.generate();
+		this.nrOfPanels = Math.ceil(items.length / itemsPerPanel);
+		this.stageItem = React.createRef();
+
+		this.state = {
+			translateX: -1000000,
+			currentPanel: 1,
+			isAutoPlayOn: this.props.hasAutoplay,
+		};
+	}
+
+	componentDidMount() {
+		checkProps(CAROUSEL, this.props, componentDoc);
+		this.setTranslationAmount(0);
+		this.stageWidth = this.stageItem.current.offsetWidth;
+		if (this.props.hasAutoplay) {
+			this.startAutoplay();
+		}
+	}
+
+	onNextPanelHandler = () => {
+		const next = this.state.currentPanel % this.nrOfPanels + 1;
+		this.setCurrentPanel(next, this.changeTranslationAutomatically);
+	};
+
+	onPreviousPanelHandler = () => {
+		const prev =
+			(this.state.currentPanel + this.nrOfPanels - 1) % this.nrOfPanels;
+		this.setCurrentPanel(prev, this.changeTranslationAutomatically);
+	};
+
+	onIndicatorClickHandler = (panel) => {
+		this.setCurrentPanel(panel, this.changeTranslationAutomatically);
+	};
+
+	onAutoPlayBtnClick = () => {
+		const { isAutoPlayOn } = this.state;
+		const actionToTake = isAutoPlayOn ? this.stopAutoplay : this.startAutoplay;
+
+		this.setState({
+			isAutoPlayOn: !isAutoPlayOn,
+		});
+		actionToTake();
+	};
+	setTranslationAmount = (amount, cb) => {
+		this.setState(() => ({ translateX: amount }), cb);
+	};
+
+	setCurrentPanel = (amount, cb) => {
+		this.setState(() => ({ currentPanel: amount }), cb);
+	};
+
+	startAutoplay = () => {
+		this.autoplayId = setInterval(() => {
+			if (this.canGoToNext()) {
+				this.onNextPanelHandler();
+			} else {
+				this.stopAutoplay();
+			}
+		}, this.props.autoplayInterval);
+	};
+
+	stopAutoplay = () => {
+		if (this.autoplayId) {
+			clearInterval(this.autoplayId);
+		}
+		this.setState({ isAutoPlayOn: false });
+	};
+
+	changeTranslationAutomatically = () => {
+		this.setTranslationAmount(
+			-(this.stageWidth * (this.state.currentPanel - 1))
+		);
+	};
+
+	canGoToNext = () => this.state.currentPanel < this.nrOfPanels;
+
+	canGoToPrevious = () => this.state.currentPanel > 1;
+
+	handleKeyDown = (event) => {
+		const keyDownCallbacks = {
+			[KEYS.LEFT]: () => {
+				if (this.canGoToPrevious()) {
+					this.onPreviousPanelHandler();
+				}
+			},
+			[KEYS.RIGHT]: () => {
+				if (this.canGoToNext()) {
+					this.onNextPanelHandler();
+				}
+			},
+		};
+
+		if (keyDownCallbacks[event.keyCode]) {
+			EventUtil.trapImmediate(event);
+			keyDownCallbacks[event.keyCode](event.shiftKey ? 10 : 1);
```

Comment:
What's going on here with the parameters? I'm not seeing them used in line 121 and 126.

Reply:
@interactivellama need to discuss for clarification

Reply:
I believe the line number is out of date. It's not https://github.com/salesforce/design-system-react/pull/1714/files#diff-b4ff71a0c207b08e9b6fcb834f32a53cR135

The question was: what is `event.shiftKey ? 10 : 1)` causing to occur on this line https://github.com/salesforce/design-system-react/pull/1714/files#diff-b4ff71a0c207b08e9b6fcb834f32a53cR136